src/components/InventoryV2/view/BatchDetailsModalForNursery.tsx

```diff
@@ -0,0 +1,544 @@
+  }, [availableAccessions, selectedBatch?.accession_id]);
+
+  useEffect(() => {
+    if (availableSubLocations && selectedBatch?.subLocations) {
+      setSelectedSubLocations(
+        availableSubLocations.filter((subLocation) =>
+          (selectedBatch.subLocations || []).find(
+            (batchSubLocation) => subLocation.id.toString() === batchSubLocation.subLocation_id
+          )
+        )
+      );
+    }
+  }, [availableSubLocations, selectedBatch?.subLocations]);
+
+  useEffect(() => {
+    if (timeZone !== tz.id) {
+      setTimeZone(tz.id);
+    }
+  }, [tz.id, timeZone]);
+
+  useEffect(() => {
+    setRecord((previousRecord: FormRecord): FormRecord => {
+      if (!previousRecord) {
+        return previousRecord;
+      }
+      return {
+        ...previousRecord,
+        addedDate: addedDateChanged ? previousRecord.addedDate : getTodaysDateFormatted(timeZone),
+      };
+    });
+  }, [timeZone, setRecord, addedDateChanged]);
+
+  useEffect(() => {
+    const newBatch: Partial<CreateBatchRequestPayload> = {
+      addedDate: getTodaysDateFormatted(),
+      facilityId: nurseryId,
+      speciesId: undefined,
+      germinatingQuantity: undefined,
+      notReadyQuantity: undefined,
+      readyQuantity: undefined,
+    };
+    const initBatch = () =>
+      selectedBatch
+        ? convertSelectedBatchToFormRecord(selectedBatch)
+        : {
+            ...newBatch,
+            id: -1,
+            batchNumber: '',
+            latestObservedTime: '',
+            version: 0,
+          };
+
+    setRecord(initBatch());
+  }, [selectedBatch, nurseryId, setRecord, selectedOrganization]);
+
+  const MANDATORY_FIELDS = [
+    'facilityId',
+    'germinatingQuantity',
+    'notReadyQuantity',
+    'readyQuantity',
+    'addedDate',
+  ] as const;
+  type MandatoryField = (typeof MANDATORY_FIELDS)[number];
+
+  const hasErrors = () => {
+    if (record) {
+      return MANDATORY_FIELDS.some((field: MandatoryField) => record[field] === '' || record[field] === undefined);
+    }
+    return true;
+  };
+
+  const saveBatch = async () => {
+    if (record) {
+      if (hasErrors()) {
+        setValidateFields(true);
+        return;
+      }
+
+      let response: (Response & BatchData) | (Response & BatchId) | undefined;
+      let responseQuantities: Partial<Response> = { requestSucceeded: true, error: undefined };
+
+      if (record.id === -1) {
+        if (record.accessionId && isBatch(record)) {
+          const nurseryTransferRecord: NurseryTransfer = {
+            date: getTodaysDateFormatted(timeZone),
+            destinationFacilityId: record.facilityId,
+            germinatingQuantity: record.germinatingQuantity,
+            notReadyQuantity: record.notReadyQuantity,
+            notes: record.notes,
+            readyByDate: record.readyByDate,
+            readyQuantity: record.readyQuantity,
+            // TODO figure out if I need this or if is auto populated by auth
+            // withdrawnByUserId: 0
```

Comment:
you can populate this from the active user, assuming this refers to the logged in user..
```
import { useUser } from 'src/providers';

...
const { user } = useUser();
...
 withdrawnByUserId: user.id,
```

Reply:
I _think_ I can actually drop this line because of https://github.com/terraware/terraware-server/blob/main/src/main/kotlin/com/terraformation/backend/seedbank/api/TransfersController.kt#L57, which would be the ideal behavior. What do you think?

Reply:
Good idea looking up the controller, yes sounds good 👍